homeassistant/components/fritzbox_callmonitor/sensor.py

```diff
@@ -20,97 +20,125 @@
-            username=username,
-            password=password,
-            phonebook_id=phonebook_id,
-            prefixes=prefixes,
+async def async_setup_platform(hass, config, async_add_entities, discovery_info=None):
+    """Import the platform into a config entry."""
+    hass.async_create_task(
+        hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_IMPORT}, data=config
         )
-    except:  # noqa: E722 pylint: disable=bare-except
-        phonebook = None
-        _LOGGER.warning("Phonebook with ID %s not found on Fritz!Box", phonebook_id)
+    )
+
 
-    sensor = FritzBoxCallSensor(name=name, phonebook=phonebook)
+async def async_setup_entry(hass, config_entry, async_add_entities):
+    """Set up the fritzbox_callmonitor sensor from config_entry."""
+    fritzbox_phonebook = hass.data[DOMAIN][config_entry.entry_id][FRITZBOX_PHONEBOOK]
 
-    add_entities([sensor])
+    phonebook_name = config_entry.title
+    phonebook_id = config_entry.data[CONF_PHONEBOOK]
+    prefixes = config_entry.options.get(CONF_PREFIXES)
+    serial_number = config_entry.data[SERIAL_NUMBER]
+    host = config_entry.data[CONF_HOST]
+    port = config_entry.data[CONF_PORT]
 
-    monitor = FritzBoxCallMonitor(host=host, port=port, sensor=sensor)
-    monitor.connect()
+    name = f"{fritzbox_phonebook.fph.modelname} Call Monitor {phonebook_name}"
+    unique_id = f"{serial_number}-{phonebook_id}"
 
-    def _stop_listener(_event):
-        monitor.stopped.set()
+    sensor = FritzBoxCallSensor(
+        name=name,
+        unique_id=unique_id,
+        fritzbox_phonebook=fritzbox_phonebook,
+        prefixes=prefixes,
+        host=host,
+        port=port,
+    )
 
-    hass.bus.listen_once(EVENT_HOMEASSISTANT_STOP, _stop_listener)
+    hass.bus.async_listen_once(
+        EVENT_HOMEASSISTANT_STOP, sensor.async_will_remove_from_hass()
+    )
 
-    return monitor.sock is not None
+    async_add_entities([sensor])
 
 
 class FritzBoxCallSensor(Entity):
     """Implementation of a Fritz!Box call monitor."""
 
-    def __init__(self, name, phonebook):
+    def __init__(self, name, unique_id, fritzbox_phonebook, prefixes, host, port):
         """Initialize the sensor."""
-        self._state = VALUE_DEFAULT
+        self._state = STATE_IDLE
         self._attributes = {}
-        self._name = name
-        self.phonebook = phonebook
+        self._name = name.title()
+        self._unique_id = unique_id
+        self._fritzbox_phonebook = fritzbox_phonebook
+        self._prefixes = prefixes
+        self._host = host
+        self._port = port
+        self._monitor = None
+
+    async def async_added_to_hass(self):
+        """Connect to FRITZ!Box to monitor its call state."""
+        _LOGGER.debug("Starting monitor for: %s", self.entity_id)
+        self._monitor = FritzBoxCallMonitor(
+            host=self._host,
+            port=self._port,
+            sensor=self,
+        )
+        self._monitor.connect()
```

Comment:
The connect does blocking I/O by creating and connecting a socket. This needs to run in the executor via `hass.async_add_executor_job`.